### DISCUSSION TITLE
Connection lost after 2-3 days

### DISCUSSION THREAD

**Discussion by Tagiwara** [MartijnvanGeffen13/Volvo4evcc#34] [Q&A]
Hi its Tagiwara from the EVCC Discussion Section.

You told me there, that i need to update to v1.8

```
Hi @Tagiwara

Can you update to 1.8.1 if using a single car, or switch to the beta of the multiple car support branch,

I think your running one of the earlier builds that had a timeing issue sometimes expiring he token to soon.

Also note after upgrade make sure the path in the start scripts match the folders you used to install (Note the Capital V of the module name, this is fixed in the latest build unless you use own folders.)

I am running with cronjob only restarting once a day and been running for over 1,5 week without the need to redo the token MFA. (In theory it can last upto 90 days before you need to reauth.)
```

a few simple questions: how would i do that? :-) and which version is running at the moment? how can i check it?

**Comment by MartijnvanGeffen13** [comment]
> Hi its Tagiwara from the EVCC Discussion Section.
> 
> You told me there, that i need to update to v1.8
> 
> ```
> Hi @Tagiwara
> 
> Can you update to 1.8.1 if using a single car, or switch to the beta of the multiple car support branch,
> 
> I think your running one of the earlier builds that had a timeing issue sometimes expiring he token to soon.
> 
> Also note after upgrade make sure the path in the start scripts match the folders you used to install (Note the Capital V of the module name, this is fixed in the latest build unless you use own folders.)
> 
> I am running with cronjob only restarting once a day and been running for over 1,5 week without the need to redo the token MFA. (In theory it can last upto 90 days before you need to reauth.)
> ```
> 
> a few simple questions: how would i do that? :-) and which version is running at the moment? how can i check it?

Hi and thx for reaching out,

For the version easiest way is to look at the Volvo4evcc.psd1 file it has the version number in it.

For the single car version we are running 1.8.1 and for the multiple car branch its 1.10.1

![image](https://github.com/user-attachments/assets/c275f25b-7f68-477a-a74e-c972ef4032bb)

To update ,

remove the volvo4evcc folder and rerun the installation instructions from the wiki ,

or if you know how to just copy and past the current version over the existing one and delete all the XML files and rerun set-volvoconfig , Do to all the new updates the old version xmls are not compatible with new versions.

**Reply by Tagiwara** [reply]
ok, got it. v1.8.1 ist installed... and i think its running, but when i start it, i reveive a error message

```
tagiwara@volvo2evcc:~$ pwsh
PowerShell 7.4.6
PS /home/tagiwara> ./Volvo4evcc/start.ps1
Token loaded from Disk succesfull
Startup with Connected:True - Charging:True
Compare-Object: /home/tagiwara/Volvo4evcc/functions/functions.ps1:218
Line |
 218 |  … eCompare = Compare-Object -ReferenceObject $LastPulseEvccData -Differ …
     |                                               ~~~~~~~~~~~~~~~~~~
     | Cannot bind argument to parameter 'ReferenceObject' because it is null.
Just a Evcc pull and token test no action taken - Token valid for another : 20.7 minutes
Just a Evcc pull and token test no action taken - Token valid for another : 20.4 minutes
Just a Evcc pull and token test no action taken - Token valid for another : 20.2 minutes
```

but i think it is ok. lets have a look in 20 Minutes :-)

**Comment by Tagiwara** [comment]
nothing changes. v1.8.1 is installed now, but i need to manually update the 6-digits token every 2-3 days

**Reply by MartijnvanGeffen13** [reply]
Currently i am on 9 days in sinds last token refresh, 

Just to be sure the logs dont show a time gap or anything indicating something else crashes or causes network issues.

What i can do is add some debug code around the token refresh to see what the reason is why your refresh is not happening . 
I will see if we can do that this week

**Reply by Tagiwara** [reply]
ok, i´ll show you what i did:

1. i start the script in the powershell with

`./Volvo4evcc/start.ps1`

2. i look at my mailbox to get the 6-digits number

3. i edit the volvo4evccconfig.xml and put in the 6-digits-number by replaycing the 111111

![Bildschirmfoto 2024-12-11 um 18 49 39](https://github.com/user-attachments/assets/82774a64-88f2-4b01-9435-31c753815a1c)

4. save the volvo4evccconfig.xml

5. after a few seconds this happends but it is running and everthing works

![Bildschirmfoto 2024-12-11 um 18 51 38](https://github.com/user-attachments/assets/9fac1ec7-1a2a-428e-afe9-d13a25d536c8)

6. after 2-3 days e get a lot of emails from volvo wit a new 6-digits-number and i have to start by 1.

**Reply by MartijnvanGeffen13** [reply]
can you show what the volvo4evcc.log file shows when the emails start to come in , 

I would expect a error in the log file at that moment when the code decides to reinitiate a full token refresh with MFA. 

Just like the happy flow example below i would expect a Failed flow in the logs for you
![image](https://github.com/user-attachments/assets/f0110034-1722-4cc8-a71e-f72b59390945)

The error above i will handle in the next update its a minor on first start as it cant compare data yet because it just started .

**Reply by MartijnvanGeffen13** [reply]
i have added some new beta code around the token refresh process to see if we can do multiple attempts when 1 fails , its in the beta branch and can be in place upgraded without making change just overwrite the module and wait till cronjob restarts or restart manually by adjusting the task)

If it keeps failing i will need to see the logs at the moment it is failing

**Comment by MartijnvanGeffen13** [comment]
Do you still run into this issue ?

I did not hear or see the logfiles from you,  for myself i am now running almost 5 weeks already without MFA reprompt
